enhancements/image-references.md

```diff
@@ -0,0 +1,228 @@
+---
+title: image-references-in-operator-bundles
+authors:
+  - "@stevekuznetsov"
+reviewers:
+  - "@jwforres"
+  - "@shawn-hurley"
+  - "@gallettilance"
+  - "@lui"
+approvers:
+  - "@jwforres"
+  - "@shawn-hurley"
+creation-date: 2020-05-17
+last-updated: 2020-05-17
+status: implementable
+see-also:
+  - "/enhancements/olm/operator-bundle.md"
+---
+
+# Image References in Operator Bundle Manifests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Manifests that make up an Operator Bundle for installation via the Operator
+Lifecycle Manager refer to one or more container images by their pull
+specifications: these container images define the operator and operands that
+the manifests deploy and manage. As with OpenShift release payloads, operator
+bundles must refer to images by digest in order to produce reproducible
+installations. A shared process to build operator bundles that replaces image
+references with fully-resolved pull specifications that pin images by digest
+must be built; this process must allow for a number of separate build systems
+to direct how these replacements occur in order to support a full-featured
+build and test strategy.
+
+## Motivation
+
+### Goals
+
+- there is one, canonical, method for building an operator bundle image layer
```

Comment:
Is this desirable, considering that we want the bundle format to be an open standard?

Reply:
Why not?

Reply:
Counter question: is there a good reason to say that `operator-sdk` is the only tool that should build bundle images (and/or image layers, which are proposed as an addition to the spec by this EP)? The spec is well-documented and there's a KEP, so any tool should be able to build a bundle image.

I'm effectively saying that this goal could be removed without any other changes to the EP.

Reply:
Perhaps this could be narrowed to specify that there is one such process for any bundles created within Red Hat build systems or productized pipelines. The standard and output format continues to be open and if someone else were to want to implement a separate process for building such an image layer, that should continue to be possible. From a consensus/maintenance perspective, however, Red Hat should have no interest in seeing separate processes exist to create bundles in our various internal build systems.

Reply:
I agree with that, but this is an upstream enhancement proposal and not specific to Red Hat so this goal should be specified elsewhere.

Reply:
What if we change `s/one/a/` in this line? I explicitly do want this to be a canonical method.

Reply:
(and I think if we do not produce something that is trivial to operate with and does not become ~ubiquitous we have failed)